code/go/0chain.net/chaincore/chain/handler.go

```
@@ -1382,7 +1386,14 @@ func PutTransaction(ctx context.Context, entity datastore.Entity) (interface{},
 	if cstate.ErrInvalidState(err) {
 		// put txn to pool if the miner got 'node not found', we should not ignore the txn because
 		// of the 'error' of the miner itself.
-		return transaction.PutTransaction(ctx, txn)
+		txnRsp, err := transaction.PutTransaction(ctx, txn)
+		if err != nil {
+			logging.Logger.Error("failed to save transaction",
+				zap.Error(err),
+				zap.Any("txn", txn))
+			return nil, common.NewErrInternal("failed to save transaction")
```

Shouldn't we return the actual error (err.Error()) as well?